gap/oper.gi

```diff
@@ -1841,3 +1841,38 @@ function(gr, edges)
   od;
   return true;
 end);
+
+InstallMethod(HamiltonianPath,
+"for a digraph",
+[IsDigraph],
+function(gr)
+  local map, obviouspath, out, i;
+
+  if DigraphNrVertices(gr) <= 1 and IsEmptyDigraph(gr) then
+    if DigraphNrVertices(gr) = 0 then
+      return [];
+    else
+      return [1];
+    fi;
+  elif not IsStronglyConnectedDigraph(gr) then
+    return fail;
+  fi;
+
+  out := OutNeighbours(gr);
+  obviouspath := true;
```

Comment:
Is it worth doing this? I think it probably isn't. Since `DigraphMonomorphism` will do this anyway.